### PR TITLE
implemented movie_sessions

### PR DESCRIPTION
**Pull request by mariiahorbova**

### CODE REVIEW

File: services/movie.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+def get_movies(
+        genres_ids: Optional[list] = None,
+        actors_ids: Optional[list] = None
+) -> QuerySet:
+
+    if genres_ids and actors_ids:
+        return Movie.objects.filter(
+            Q(genres__pk__in=genres_ids)
+            & Q(actors__pk__in=actors_ids)
```
Comment (oleksiikolii) [inline services/movie.py:16]: You can remove this check if you put query into variable and than apply filters to this query if needed, eventually return this query

File: services/movie_session.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    )
+
+
+def get_movies_sessions(
+        session_date: Optional[str] = None
+) -> QuerySet:
+    movies_sessions = MovieSession.objects.all()
+    if session_date:
+        return MovieSession.objects.filter(
+            show_time__date=session_date
```
Comment (oleksiikolii) [inline services/movie_session.py:25]: Same here, you can do `movies = movies_sessions.filter(....)` and use one return in the end

**Review by oleksiikolii — CHANGES_REQUESTED** [review]

**Review by oleksiikolii — APPROVED** [review]
Good job!

File: db/models.py [hunk 3]
```diff
@@ -14,3 +14,41 @@ class Actor(models.Model):
...
+    def __str__(self) -> str:
+        return self.title
+
+
+class CinemaHall(models.Model):
+    name = models.CharField(max_length=255)
+    rows = models.IntegerField()
+    seats_in_row = models.IntegerField()
+
+    @property
```
Comment (LiudmylaKulinchenko) [inline db/models.py:34]: using the decorator is not necessary. You can remove it

File: services/cinema_hall.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+from django.db.models import QuerySet
+
+from db.models import CinemaHall
+
+
+def get_cinema_halls() -> QuerySet:
+    return CinemaHall.objects.all()
+
+
+def create_cinema_hall(hall_name: str,
```
Comment (LiudmylaKulinchenko) [inline services/cinema_hall.py:10]: fix indentation here

File: services/movie_session.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
+from typing import Optional
+
+from django.db.models import QuerySet
+
+from db.models import CinemaHall, Movie, MovieSession
+
+
+def create_movie_session(movie_show_time: str,
```
Comment (LiudmylaKulinchenko) [inline services/movie_session.py:8]: fix indentation

File: services/movie_session.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+                         movie_id: int,
+                         cinema_hall_id: int
+                         ) -> None:
+    MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall=CinemaHall(pk=cinema_hall_id)
```
Comment (LiudmylaKulinchenko) [inline services/movie_session.py:15]: Use checklist "Code Efficiency" first point

File: services/movie_session.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    )
+
+
+def get_movies_sessions(
+        session_date: Optional[str] = None
```
Comment (LiudmylaKulinchenko) [inline services/movie_session.py:20]: If you specify a default value of `None` for a parameter, you don't need to use the `Optional` type hint

File: services/movie.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    return movies
+
+
+def get_movie_by_id(movie_id: int) -> Movie:
+    return Movie.objects.get(pk=movie_id)
+
+
+def create_movie(movie_title: str,
+                 movie_description: str,
+                 genres_ids: Optional[list] = None,
```
Comment (LiudmylaKulinchenko) [inline services/movie.py:29]: You can remove Optional and use `list[int]` instead

File: services/movie_session.py [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    if session_date:
+        movies_sessions = movies_sessions.filter(show_time__date=session_date)
+    return movies_sessions
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(pk=movie_session_id)
+
+
+def update_movie_session(session_id: int,
```
Comment (LiudmylaKulinchenko) [inline services/movie_session.py:32]: Indentation is wrong in such cases. Fix everywhere

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

File: db/models.py [hunk 10]
```diff
@@ -14,3 +14,41 @@ class Actor(models.Model):
 
     def __str__(self) -> str:
         return f"{self.first_name} {self.last_name}"
+
+
+class Movie(models.Model):
+    title = models.CharField(max_length=255)
+    description = models.TextField()
+    actors = models.ManyToManyField(Actor)
+    genres = models.ManyToManyField(Genre)
```
Comment (LiudmylaKulinchenko) [inline db/models.py:23]: Add related_name

**Review by YuriiKindrat — CHANGES_REQUESTED** [review]

**Review by YuriiKindrat — APPROVED** [review]

**Pull request closed without merging**